### PR TITLE
Challenge 2

### PR DESCRIPTION
**Pull request by lynn-thira**
Integration of UI templates developed in previous challenge into flask app
Development of all user features including interfaces

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,65 @@
...
+```
+python run.py
+```
+Then navigate to your localhost, on port 5000, like so `localhost:5000`. Go ahead and give it a try. :)
+## Running the tests
+
+To run the tests that come bundled with the app, type the following command in the cmd or a IDE terminal such as that on Vscode
+
+```
+pytest test_app.py
```
Comment (ivantoz) [inline README.md]: this script does not run test, fix this

File: tests/test_app.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        response = self.tester.get('/signup', content_type='html/text')
+        self.assertEqual(response.status_code, 200)
+
+    def test_signin_page_works(self):
+        """defines test sign in page"""
+
+        response = self.tester.get('/signin', content_type='html/text')
+        self.assertEqual(response.status_code, 200)
+
+    # def test_missing_templates_status(self):
```
Comment (ivantoz) [inline tests/test_app.py]: Why is this test commented out?

File: tests/test_user.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                entered_password = hash_object.hexdigest()
+
+                self.assertEqual(models.all_users['jb@email.com'][2],
+                                 '3c01bdbb26f358bab27f267924aa2c9a03fcfdb8',
+                                 "User cannot be authenticated")
+
+
+
+
+        # assert b"The Checklist | Signin" in response1.data
```
Comment (ivantoz) [inline tests/test_user.py]: Why is this commented out?

File: tests/test_app.py (hunk 2 above)
Comment (lynn-thira) [reply tests/test_app.py]: I was doing a testing on missing templates, I'm still working on implementing this under the models, shall update and test as soon as it runs.

File: tests/test_user.py (hunk 3 above)
Comment (lynn-thira) [reply tests/test_user.py]: I got a pylint error in relation to this, work around required changing the test case test_user. One more functionality to change under app before I can uncomment and test

**Pull request merged**